pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -430,6 +438,164 @@ func consumerRefMatches(consumer *corev1.ObjectReference, machine *machinev1.Mac
+		if err != nil {
+			log.Printf("Error base64 decoding userData : %s", err.Error())
+			return fmt.Errorf("error base64 decoding userData: %w", err)
+		}
+
+		// Define an ignition data struct to unmarshal incoming json
+		type IgnitionData struct {
+			Ignition struct {
+				Config struct {
+					Append []struct {
+						Source string
+					}
+				}
+				Security struct {
+					TLS struct {
+						CertificateAuthorities []struct {
+							Source string
+						}
+					}
+				}
+			}
+		}
+		var ignData IgnitionData
+
+		if err := json.Unmarshal(userData, &ignData); err != nil {
+			log.Printf("Cannot unmarshal ignition config from userData : %s, %w", config.UserData.Name, err)
+			return err
+		}
+
+		var ignitionURL, caCertRaw string
+		// Fetch IgnitionURL and the CA cert
+		if len(ignData.Ignition.Config.Append) > 0 && len(ignData.Ignition.Config.Append[0].Source) > 0 {
+			ignitionURL = ignData.Ignition.Config.Append[0].Source
+		} else {
+			// We didn't get any URL
+			err := fmt.Errorf("error extracting full Ignition URL")
+			log.Printf("URL not found : %s", err.Error())
+			return err
+		}
+
+		if len(ignData.Ignition.Security.TLS.CertificateAuthorities) > 0 && len(ignData.Ignition.Security.TLS.CertificateAuthorities[0].Source) > 0 {
+			caCertRaw = ignData.Ignition.Security.TLS.CertificateAuthorities[0].Source
+		} else {
+			// We didn't get the URL's Cert
+			err := fmt.Errorf("CA cert for full Ignition URL was not specified")
+			log.Printf("Cert not found in ignition data: %s", err.Error())
+			return err
+		}
+		caCertB64 := strings.TrimPrefix(caCertRaw, "data:text/plain;charset=utf-8;base64,")
+
+		caCert, err := base64.StdEncoding.DecodeString(caCertB64)
+		if err != nil {
+			log.Printf("Could not base64 decode caCert: %s", err.Error())
+			return fmt.Errorf("could not base64 decode caCert: %w", err)
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+		client := retryablehttp.NewClient()
+		client.HTTPClient.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: caCertPool,
+			},
+		}
+
+		// Get the full ignition from the URL
+		resp, err := client.Get(ignitionURL)
+		if err != nil || resp.StatusCode != http.StatusOK {
+			log.Printf("Error fetching full ignition from URL : %s", err.Error())
+			return fmt.Errorf("error fetching full ignition from URL: %w", err)
+		}
+		defer resp.Body.Close()
+
+		var fullIgnition []byte
+		fullIgnition, err = ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Printf("Error reading full ignition data from URL : %s", err.Error())
+			return fmt.Errorf("error reading full ignition data from URL: %w", err)
+		}
+
+		// Create a new secret with ignition data
+		machineUserDataSecret = createNewMachineSecret(fullIgnition, machine)
+		err = a.client.Create(ctx, machineUserDataSecret)
+		if err != nil {
+			log.Printf("Cannot create the %s secret: %s", machine.Name+"-user-data", err.Error())
```

Comment:
If both createNewMachineSecret() and this error handling code know how to build the name, then we have to change 2 places if we ever change the naming. We have access to the name in the return value from createNewMachineSecret, so let's use that.

```suggestion
			log.Printf("Cannot create the %s secret: %s", machineUserDataSecret.Name, err.Error())
```